comment/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+#![allow(clippy::or_fun_call)]
+use std::ffi::OsString;
+use std::str::FromStr;
+
+use anyhow::anyhow;
+
+use radicle_common::args::{Args, Error, Help};
+use radicle_common::{
+    cobs::{self, issue::*},
+    keys, person, profile, project,
+};
+use radicle_terminal as term;
+
+pub const HELP: Help = Help {
+    name: "comment",
+    description: env!("CARGO_PKG_DESCRIPTION"),
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad comment <id> [--description <text>]
+
+Options
+
+    --help      Print help
+"#,
+};
+
+#[derive(Debug)]
+pub struct Options {
+    pub id: cobs::issue::IssueId,
+    pub description: Option<String>,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut id: Option<cobs::issue::IssueId> = None;
+        let mut description: Option<String> = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Long("description") => {
+                    description = Some(parser.value()?.to_string_lossy().into());
+                }
+                Value(val) if id.is_none() => {
+                    let val = val
+                        .to_str()
+                        .ok_or_else(|| anyhow!("issue id specified is not UTF-8"))?;
+
+                    id = Some(
+                        IssueId::from_str(val)
+                            .map_err(|_| anyhow!("invalid issue id '{}'", val))?,
+                    );
+                }
+                _ => {
+                    return Err(anyhow!(arg.unexpected()));
+                }
+            }
+        }
+
+        Ok((
+            Options {
+                id: id.ok_or_else(|| anyhow!("an issue id must be provided"))?,
+                description,
+            },
+            vec![],
+        ))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let profile = profile::default()?;
+    let signer = term::signer(&profile)?;
+    let storage = keys::storage(&profile, signer)?;
+    let (project, _) = project::cwd()?;
+    let whoami = person::local(&storage)?;
+    let issues = Issues::new(whoami.clone(), profile.paths(), &storage)?;
+    let patches = cobs::patch::Patches::new(whoami, profile.paths(), &storage)?;
+    let id = options.id;
+
+    let doc = options
+        .description
+        .unwrap_or("Enter a description...".to_owned());
+
+    if let Some(text) = term::Editor::new().edit(&doc)? {
+        if let Some(_issue) = issues.get(&project, &id)? {
+            issues.comment(&project, &id, &text)?;
+        } else if let Some(patch) = patches.get(&project, &id)? {
+            patches.update(&project, &id, &text, patch.revisions.last().oid)?;
```

Comment:
This doesn't add a comment, this adds a revision. We need to add a `comment` function to `Patches`

Reply:
https://github.com/radicle-dev/radicle-cli/pull/146/files#diff-917043cf94ba716b366ad78c27eced853818479cd52801c938b59bea571bc81eR112-R119